### PR TITLE
ToString for ApiException should contain response payload

### LINKED ISSUE #945: when exception raised, ToString should contain response payload
This will help for debugging scenarios where an intermittent error occurs.

Originally raised in #944

### PR DESCRIPTION
**Pull request by asizikov**
Overrided `ToString` methods for `ApiException` and `ApiError`
Fixes #945 

I'm not really sure about format though. Right now `ToString` produces the following output: 

```
Octokit.ApiException: Validation Failed
Octokit.ApiErrorMessage: Validation Failed
Documentation url: 
Errors:
Message: key is already in use
Code: custom
Field: key
Resource: PublicKey
```

- [x] agree on message format
- [x] update current message format if needed

### CODE REVIEW

**Comment by asizikov** [comment]
@shiftkey could you please share your vision about the format of message?

**Comment by Haacked** [comment]
I think we should just show the Response payload. It'll omit the fields that don't exist.

**Comment by asizikov** [comment]
@Haacked so I just should store raw unparsed payload in the `ApiException`? Ok, will update the PR tomorrow. Thanks for quick reply!

**Comment by Haacked** [comment]
@asizikov Yeah, if that's easy. @shiftkey what do you think?

**Comment by shiftkey** [comment]
@Haacked @asizikov  make it so!

**Comment by asizikov** [comment]
Hmm, the build seems to be broken. I guess it's not related to my changes.

Anyway, here is the new output sample: 

```
Octokit.ApiException: Validation Failed
{"errors":[{"code":"custom","field":"key","message":"key is already in use","resource":"PublicKey"}],"message":"Validation Failed"}
```

**Comment by shiftkey** [comment]
@asizikov yeah, that's #983

**Comment by asizikov** [comment]
oh, cool, you're aware of it :+1:

**Comment by shiftkey** [comment]
@asizikov if you merge `master` into your branch you should have a green build again

File: Octokit/Models/Response/ApiError.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Collections.Generic;
+using System.Text;
```
Comment (shiftkey) [inline Octokit/Models/Response/ApiError.cs]: :lipstick: this namespace doesn't seem necessary

Comment (asizikov) [inline Octokit/Models/Response/ApiError.cs]: yup, it doesn't. 
Deleted.

**Comment by asizikov** [comment]
Aaand it's green now!

**Comment by shiftkey** [comment]
@asizikov thanks!

**Pull request merged**